neofs-cli object hash leads to 'panic: runtime error: invalid memory address or nil pointer dereference'

## Steps to Reproduce (for bugs)
1. Create container
```
neofs-cli --rpc-endpoint 172.26.162.50:8080 --wallet wallet.json container create --name f_test_1 --policy "REP 2 IN X CBF 2 SELECT 2 FROM * AS X" --await
Enter password >
container ID: 2tE4gM4zVdk3StGC24Yd3LPXL4bxNvCjzqcxUdqNnRdk
```
2. PUT object
```
neofs-cli --rpc-endpoint 172.26.162.50:8080 -w wallet.json object put --file /tmp/object100M-1.sample --cid 2tE4gM4zVdk3StGC24Yd3LPXL4bxNvCjzqcxUdqNnRdk
Enter password >
 104857600 / 104857600 [===========================================================================================================================================================================] 100.00% 24s
[/tmp/object100M-1.sample] Object successfully stored
  OID: 2SDJJ9cwoZteddtdEQ7TNQ41AJvm7yK57AnjLCtN846L
  CID: 2tE4gM4zVdk3StGC24Yd3LPXL4bxNvCjzqcxUdqNnRdk
```
3. Try to get hash of the object
```
neofs-cli object hash --rpc-endpoint '172.26.162.50:8080' --wallet 'wallet.json' --cid '2tE4gM4zVdk3StGC24Yd3LPXL4bxNvCjzqcxUdqNnRdk' --oid '2SDJJ9cwoZteddtdEQ7TNQ41AJvm7yK57AnjLCtN846L' --type sha256
Enter password >
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0xa502ac]

goroutine 1 [running]:
github.com/nspcc-dev/neofs-sdk-go/client.(*Client).prepareRequest(0x0?, {0xe60aa0?, 0xc000382060?}, 0xc00033eb40?)
        github.com/nspcc-dev/neofs-sdk-go@v1.0.0-rc.7/client/common.go:177 +0xac
github.com/nspcc-dev/neofs-sdk-go/client.(*Client).ObjectHead(0x0, {0xe620a0?, 0xc00013c000}, {{{0x0, 0x0, 0x0, {0x0, 0x0, 0x0}, 0x0, ...}, ...}, ...})
        github.com/nspcc-dev/neofs-sdk-go@v1.0.0-rc.7/client/object_get.go:433 +0x15f
github.com/nspcc-dev/neofs-node/cmd/neofs-cli/internal/client.HeadObject({{{0x0}, {0x0}, 0x0, 0x0, {0x14bff20, 0x0, 0x0}}, {{{0x1b, 0xfb, 0x14, ...}, ...}}, ...})
        github.com/nspcc-dev/neofs-node/cmd/neofs-cli/internal/client/client.go:621 +0x3df
github.com/nspcc-dev/neofs-node/cmd/neofs-cli/modules/object.getObjectHash(0x147c700?, {0xd15baa?, 0xa?, 0xa?})
        github.com/nspcc-dev/neofs-node/cmd/neofs-cli/modules/object/hash.go:75 +0x2fd
github.com/spf13/cobra.(*Command).execute(0x147c700, {0xc00033f220, 0xa, 0xa})
        github.com/spf13/cobra@v1.5.0/command.go:876 +0x67b
github.com/spf13/cobra.(*Command).ExecuteC(0x1476d00)
        github.com/spf13/cobra@v1.5.0/command.go:990 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v1.5.0/command.go:918
github.com/nspcc-dev/neofs-node/cmd/neofs-cli/modules.Execute()
        github.com/nspcc-dev/neofs-node/cmd/neofs-cli/modules/root.go:52 +0x25
main.main()
        github.com/nspcc-dev/neofs-node/cmd/neofs-cli/main.go:6 +0x17
```
**Version:**
```
NeoFS CLI
Version: v0.34.0-9-g2e891768
GoVersion: go1.18.4
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -37,6 +37,7 @@ Changelog for NeoFS Node
 - Write-cache flush duplication (#2074)
 - Ignore error if a transaction already exists in a morph client (#2075)
 - Pack arguments of `setPrice` invocation during contract update (#2078)
+- `neofs-cli object hash` panic (#2079)
 
 ### Removed
 ### Updated

--- a/cmd/neofs-cli/modules/object/hash.go
+++ b/cmd/neofs-cli/modules/object/hash.go
@@ -64,11 +64,13 @@ func getObjectHash(cmd *cobra.Command, _ []string) {
 	common.ExitOnErr(cmd, "could not decode salt: %w", err)
 
 	pk := key.GetOrGenerate(cmd)
+	cli := internalclient.GetSDKClientByFlag(cmd, pk, commonflags.RPC)
 
 	tz := typ == hashTz
 	fullHash := len(ranges) == 0
 	if fullHash {
 		var headPrm internalclient.HeadObjectPrm
+		headPrm.SetClient(cli)
 		Prepare(cmd, &headPrm)
 		headPrm.SetAddress(objAddr)
 
@@ -93,8 +95,6 @@ func getObjectHash(cmd *cobra.Command, _ []string) {
 
 		return
 	}
-
-	cli := internalclient.GetSDKClientByFlag(cmd, pk, commonflags.RPC)
 
 	var hashPrm internalclient.HashPayloadRangesPrm
 	hashPrm.SetClient(cli)

